### DISCUSSION TITLE
Is it possible to complie deno code to rust code?

### DISCUSSION THREAD

**Discussion by ghost** [denoland/deno#13015] [Ideas]
Is it possible to compile deno to pure rust code as it is built on top of rust?

The benefit is we could deploy the "rust" code on AWS Lambda without long cold start time.

**Comment by lucacasonato** [comment]
No, this is not possible. You can not compile a dynamically typed, type unsound, garbage collected language to a sound statically typed language like Rust.

**Reply by jihaohaaaa** [reply]
Actually, Typescript's type system is more complicated than Rust (even c/c++). The really difficult transforming from ts to rs is ts's (v8) garbage collect system.
